tests/test_output_parser.py

```diff
@@ -41,7 +41,40 @@ def test_output_parser_returns_error_when_model_output_returns_different_field(s
         # WHEN & THEN
         for model in models.values():
             model_response = model.generate(prompt, input_data, SummaryOutputClass)
-            assert type(model_response[0].error) == OutputParserException
+            assert "OutputParserException" in model_response[0].error
+            assert model_response[0].response is None
+
+    @patch("langchain.chains.base.Chain.arun")
+    @patch("langchain_community.llms.vertexai.VertexAI.get_num_tokens")
+    def test_output_parser_extracts_json_from_response(self, tokens_mock, chain_run_mock, models):
+        # GIVEN
+        text_output = "This is the model output"
+        expected_model_response = json.dumps({"summary": text_output})
+        chain_run_mock.return_value = f"Sure! Here's the JSON you wanted: {expected_model_response} Have a nice day!"
+        tokens_mock.return_value = 1
+
+        input_data = [InputData(input_mappings={"text": "Some dummy text"}, id="1")]
+        prompt = "Some Dummy Prompt {text}"
+
+        # WHEN & THEN
+        for model in models.values():
+            model_response = model.generate(prompt, input_data, SummaryOutputClass)
+            assert model_response[0].response == SummaryOutputClass(summary=text_output)
+
+    @patch("langchain.chains.base.Chain.arun")
+    @patch("langchain_community.llms.vertexai.VertexAI.get_num_tokens")
+    def test_output_parser_returns_error_when_json_is_garbled(self, tokens_mock, chain_run_mock, models):
```

Comment:
From my experience, models sometimes outputs a json in unpredictable formats. For example:

```
{"output": "Sure, here is the output for the given product review in valid JSON format:\n\n{\n\"score\": 0\n}"},
{"output": "<<SYS>>\\n{\\n    \"score\": 0\\n}\\n<</SYS>>"},
{"output": "{\\\"score\\\": 0}\\n"}
```

Will such cases pass the test? Maybe it's better to add them.
If the test is not passed, we require a normalization step:

```
def normalize_string_before_parsing_to_json(input_str: str) -> str:
    input_without_new_line_chars = input_str.replace("\\n", "")
    input_without_escape_chars = input_without_new_line_chars.replace("\\", "")
    return input_without_escape_chars
```

Reply:
I've added this normalization logic + respective test cases.